worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/DefaultBlockParser.java

```diff
@@ -277,91 +279,79 @@ private BaseBlock parseLogic(String input, ParserContext context) throws InputPa
+                    return blockInHand;
+                }
+
+                blockType = blockInHand.getBlockType();
+                blockStates.putAll(blockInHand.getStates());
             } else if ("offhand".equalsIgnoreCase(typeString)) {
                 // Get the block type from the item in the user's off hand.
                 final BaseBlock blockInHand = getBlockInHand(context.requireActor(), HandSide.OFF_HAND);
-                state = blockInHand.toBlockState();
-                nbt = blockInHand.getNbtData();
-            } else if (typeString.matches("pos[0-9]+")) {
-                int index = Integer.parseInt(typeString.replaceAll("[a-z]+", ""));
+                if (blockInHand.getClass() != BaseBlock.class) {
+                    return blockInHand;
+                }
+
+                blockType = blockInHand.getBlockType();
+                blockStates.putAll(blockInHand.getStates());
+            } else if ("pos1".equalsIgnoreCase(typeString)) {
                 // Get the block type from the "primary position"
                 final World world = context.requireWorld();
                 final BlockVector3 primaryPosition;
                 try {
-                    primaryPosition = context.requireSession().getRegionSelector(world).getVertices().get(index - 1);
+                    primaryPosition = context.requireSession().getRegionSelector(world).getPrimaryPosition();
                 } catch (IncompleteRegionException e) {
-                    throw new InputParseException("Your selection is not complete.");
+                    throw new InputParseException(TranslatableComponent.of("worldedit.error.incomplete-region"));
                 }
-                state = world.getBlock(primaryPosition);
-            } else if (typeString.matches("slot[0-9]+")) {
-                int slot = Integer.parseInt(typeString.substring(4)) - 1;
-                Actor actor = context.requireActor();
-                if (!(actor instanceof Player)) {
-                    throw new InputParseException("The user is not a player!");
-                }
-                Player player = (Player) actor;
-                BlockBag bag = player.getInventoryBlockBag();
-                if (!(bag instanceof SlottableBlockBag)) {
-                    throw new InputParseException("Unsupported!");
-                }
-                SlottableBlockBag slottable = (SlottableBlockBag) bag;
-                BaseItem item = slottable.getItem(slot);
+                final BlockState blockInHand = world.getBlock(primaryPosition);
 
-                if (!item.getType().hasBlockType()) {
-                    throw new InputParseException("You're not holding a block!");
-                }
-                state = item.getType().getBlockType().getDefaultState();
-                nbt = item.getNbtData();
+                blockType = blockInHand.getBlockType();
+                blockStates.putAll(blockInHand.getStates());
```

Comment:
Don't start deleting stuff. here. You removed 2 features (one may not be functional anyways). Please revert.